airbyte-cdk/python/unit_tests/sources/declarative/migrations/test_legacy_to_per_partition_migration.py

```diff
@@ -0,0 +1,289 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+from airbyte_cdk.sources.declarative.migrations.legacy_to_per_partition_state_migration import LegacyToPerPartitionStateMigration
+from airbyte_cdk.sources.declarative.models import (
+    CustomRetriever,
+    DatetimeBasedCursor,
+    DeclarativeStream,
+    ParentStreamConfig,
+    SubstreamPartitionRouter,
+)
+
+
+def test_migrate_a_valid_legacy_state_to_per_partition():
+    input_state = {
+        "13506132": {
+            "last_changed": "2022-12-27T08:34:39+00:00"
+        },
+        "14351124": {
+            "last_changed": "2022-12-27T08:35:39+00:00"
+        },
+    }
+
+    migrator = _migrator()
+
+    assert migrator.should_migrate(input_state)
+
+    expected_state = {
+        "states": [
+            {
+                "partition": {"id": "13506132"},
+                "cursor": {"last_changed": "2022-12-27T08:34:39+00:00"}
+            },
+            {
+                "partition": {"id": "14351124"},
+                "cursor": {"last_changed": "2022-12-27T08:35:39+00:00"}
+            },
+        ]
+    }
+
+    assert migrator.migrate(input_state) == expected_state
+
+
+def test_should_not_migrate_a_per_partition_state():
```

Comment:
nit, but would these work better as parameterized scenarios where input_state, and expected bool are the inputs? Seems like all the below tests follow a similar format for the test

Reply:
done!